### PR TITLE
Some OS:: cleanups/changes

### PR DESCRIPTION
**Pull request by randombit**

### CODE REVIEW

File: src/lib/utils/os_utils.cpp [hunk 1]
```diff
@@ -206,32 +206,28 @@ uint64_t OS::get_cpu_cycle_counter()
...
-   if ((res = ::sysconf(_SC_NPROCESSORS_CONF)) <= 0)
-     {
-          return 1;
-     }
-     tt = static_cast<size_t>(res);
-#elif defined(BOTAN_TARGET_OS_HAS_THREADS)
-     tt = static_cast<size_t>(std::thread::hardware_concurrency());
-#endif
-     return tt;
+   const long res = ::sysconf(_SC_NPROCESSORS_CONF);
```
Comment (devnexen) [inline src/lib/utils/os_utils.cpp]: I see you rarely use `auto` (maybe for clarity's sake) so I just followed but nothing against.

**Comment by devnexen** [comment]
Was thinking if a clang format config would be useful with this PR.

File: src/lib/utils/os_utils.cpp (hunk 1 above)
Comment (randombit) [reply src/lib/utils/os_utils.cpp]: Yeah in general I only use `auto` when the type name is either very long (`std::map<...>::const_iterator`) or is actually untypable (lambda, based on template param, ...).

**Comment by randombit** [comment]
Re code style there is an open issue #424 actually the oldest unresolved issue. It has been a while since I looked at it. What I wanted was something that gets within ~90% of the current codestyle and apply the formatter across the codebase, then we could from there consider future format changes. I was not able to get there with clang-format at the time, but I imagine it has improved in last years. There is an `astyle` template posted in that issue that gets pretty close though (also in `src/configs/astyle.rc`). But anyway I think a code reformat should happen in its own PR with no other (functional) code changes.

**Pull request merged**